src/System.Windows.Forms/src/System/Windows/Forms/CheckBox.CheckBoxAccessibleObject.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using static System.Windows.Forms.Control;
+using static Interop;
+
+namespace System.Windows.Forms
+{
+    public partial class CheckBox
+    {
+        [ComVisible(true)]
+        internal class CheckBoxAccessibleObject : ControlAccessibleObject
+        {
+            internal CheckBoxAccessibleObject(CheckBox owner) : base(owner)
+            {
+            }
+
+            private CheckBox OwningCheckBox => Owner as CheckBox;
+
+            internal override bool IsIAccessibleExSupported() => true;
+
+            internal override bool IsPatternSupported(UiaCore.UIA patternId)
+            {
+                if (patternId == UiaCore.UIA.TogglePatternId)
+                {
+                    return true;
+                }
+
+                return base.IsPatternSupported(patternId);
+            }
+
+            internal override object GetPropertyValue(UiaCore.UIA propertyID)
+            {
+                switch (propertyID)
+                {
+                    case UiaCore.UIA.NamePropertyId:
+                        return Name;
+                    case UiaCore.UIA.AutomationIdPropertyId:
+                        return OwningCheckBox.Name;
+                    case UiaCore.UIA.IsTogglePatternAvailablePropertyId:
+                        return IsPatternSupported(UiaCore.UIA.TogglePatternId);
+                    case UiaCore.UIA.ControlTypePropertyId:
+                        return UiaCore.UIA.CheckBoxControlTypeId;
+                    case UiaCore.UIA.IsKeyboardFocusablePropertyId:
+                        // This is necessary for compatibility with MSAA proxy:
+                        // IsKeyboardFocusable = true regardless the control is enabled/disabled.
+                        return true;
+                }
+
+                return base.GetPropertyValue(propertyID);
+            }
+
+            public override string DefaultAction
+            {
+                get
+                {
+                    string defaultAction = Owner.AccessibleDefaultActionDescription;
+                    if (defaultAction != null)
+                    {
+                        return defaultAction;
+                    }
+
+                    if (((CheckBox)Owner).Checked)
```

Comment:
You can use `OwningCheckBox` here to you will not need to cast it to `Checkbox`.